catalog/tests/test_views.py

```diff
@@ -0,0 +1,280 @@
+from django.test import TestCase
+from django.urls import reverse
+from django.utils import timezone
+from django.contrib.auth.models import User, Permission 
+from catalog.models import BookInstance, Book, Genre, Author
+import datetime
+import uuid
+
+
+class AuthorListViewTest(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        # Create 13 authors for pagination tests
+        number_of_authors = 13
+
+        for author_id in range(number_of_authors):
+            Author.objects.create(
+                first_name=f'Christian {author_id}',
+                last_name=f'Surname {author_id}',
+            )
+
+    def test_view_url_exists_at_desired_location(self):
+        response = self.client.get('/catalog/authors/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_url_accessible_by_name(self):
+        response = self.client.get(reverse('authors'))
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_uses_correct_template(self):
+        response = self.client.get(reverse('authors'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'catalog/author_list.html')
+
+    def test_pagination_is_ten(self):
+        response = self.client.get(reverse('authors'))
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['author_list']), 10)
+
+    def test_lists_all_authors(self):
+        # Get second page and confirm it has (exactly) remaining 3 items
+        response = self.client.get(reverse('authors')+'?page=2')
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('is_paginated' in response.context)
+        self.assertTrue(response.context['is_paginated'] == True)
+        self.assertEqual(len(response.context['author_list']), 3)
+
+
+class LoanedBookInstancesByUserListViewTest(TestCase):
+    def setUp(self):
```

Comment:
method này dùng để làm gì vậy em?

Reply:
`setUp ()` được gọi đầu mọi hàm kiểm tra để thiết lập đối tượng, cái mà có thể đã bị sửa đổi trong các test trước, nó sẽ làm mới lại object này như lúc `setUpTestData()` khởi tạo test data ạ (set up any objects that may be modified by the test (every test function will get a "fresh" version of these objects))